pkg/api/validate/dynamic/dynamic.go

```diff
@@ -607,34 +612,80 @@ func (dv *dynamic) ValidateSubnets(ctx context.Context, oc *api.OpenShiftCluster
 			)
 		}
 
-		if oc.Properties.ProvisioningState == api.ProvisioningStateCreating {
-			if ss.SubnetPropertiesFormat != nil &&
-				ss.SubnetPropertiesFormat.NetworkSecurityGroup != nil {
-				expectedNsgID, err := subnet.NetworkSecurityGroupID(oc, s.ID)
+		subnetByID[s.ID] = ss
+	}
+	return subnetByID, nil
+}
+
+// checkByoNSG checks whether all the subnets have or don't have NSG attached.
+// when the BYONsg feature flag is on and only some of the subnets are attached with an NSG,
+// it returns an error.  If none of the subnets is attached, it's no longer BYO NSG and the
+// cluster installation process should fall back to using the managed nsg.
+func (dv *dynamic) checkByoNSG(subnetByID map[string]*mgmtnetwork.Subnet) (bool, error) {
+	noNSGAttached := 0
+	allSubnetsAreAttached := len(subnetByID)
+
+	var attached int
+	for _, subnet := range subnetByID {
+		if subnet.NetworkSecurityGroup != nil && subnet.NetworkSecurityGroup.ID != nil {
```

Comment:
you wrote a helper function for this logic below.  Use that.

Reply:
That one was not the same although the name sounds like it (it's for a different type).  I have added a new one to keep things consistent anyway.